projects-to-be-tested/spring-boot/helm-config/pom.xml

```diff
@@ -0,0 +1,108 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (c) 2019 Red Hat, Inc.
+    This program and the accompanying materials are made
+    available under the terms of the Eclipse Public License 2.0
+    which is available at:
+
+        https://www.eclipse.org/legal/epl-2.0/
+
+    SPDX-License-Identifier: EPL-2.0
+
+    Contributors:
+      Red Hat, Inc. - initial API and implementation
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.eclipse.jkube.integration-tests</groupId>
+    <artifactId>jkube-integration-tests-project</artifactId>
+    <version>${revision}</version>
+    <relativePath>../../../pom.xml</relativePath>
+  </parent>
+
+  <artifactId>spring-boot-helm-config</artifactId>
+  <name>${global.name} :: Spring Boot :: Helm Config</name>
+  <description>
+    Spring Boot with Helm Config project
+  </description>
+
+  <dependencies>
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-web</artifactId>
+      <scope>compile</scope>
+    </dependency>
+  </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-maven-plugin</artifactId>
+        <executions>
+          <execution>
+            <goals>
+              <goal>repackage</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>org.eclipse.jkube</groupId>
+        <artifactId>kubernetes-maven-plugin</artifactId>
+        <configuration>
+          <helm>
+            <chart>This is the chart name</chart>
+            <chartExtension>tar</chartExtension>
+            <version>1.0-KUBERNETES</version>
+            <description>Description different to that in the pom.xml</description>
+            <home>https://www.home.example.com</home>
+            <engine>v8</engine>
+            <sources>
+              <source>https://source.1.example.com</source>
+            </sources>
+            <maintainers>
+              <maintainer>
+                <name>Mr. Ed</name>
+                <email>ed@example.com</email>
+              </maintainer>
+            </maintainers>
+          </helm>
+        </configuration>
+      </plugin>
+      <plugin>
+        <groupId>org.eclipse.jkube</groupId>
+        <artifactId>openshift-maven-plugin</artifactId>
```

Comment:
I'm not sure if OpenShift 3.v supports Helm. Right now all of our tests are at 3.x versions.

Reply:
The test generates and validates the helm chart and templates. It doesn't deploy the charts in the cluster (helm install)
I was hesitant to implement the chart deployment but I finally decided to let this one go. We have an open issue to  reimplement helm feature for helm version 3, so this can be part of that task. I've already spent too much time with the Helm refactor :(
The test that applies to this project verifies that the charts are according to our opinionated chart generation spec for OpenShift (e.g. helm tarball has a "helmishift" prefix, etc.)